controlplane/kubeadm/internal/controllers/remediation.go

```diff
@@ -311,7 +311,8 @@ func (r *KubeadmControlPlaneReconciler) reconcileUnhealthyMachines(ctx context.C
 	}
 
 	// Surface the operation is in progress.
-	log.Info("Remediating unhealthy machine")
+	log.WithValues(controlPlane.StatusToLogKeyAndValues(nil, machineToBeRemediated)...).
```

Comment:
Should this also be before the Delete call like in the "KCP deleted" case?

I think that would be good so we have the timestamp of directly before Machine deletion and not afterwards. Should be easier for cross-referencing with other logs